demos/smart_classroom_demo_gapi/main.cpp

```diff
@@ -628,90 +560,175 @@ int main(int argc, char* argv[]) {
         }
 
-        std::unique_ptr<AsyncDetection<DetectedAction>> action_detector;
+        /** NOTE: Empty strings. Need default constructor for cv::gapi::ie::Params
+          * error msg: no appropriate default constructor
+          */
+        cv::gapi::ie::Params<nets::FaceDetector> det_net({}, {}, {});
+        cv::gapi::ie::Params<nets::LandmarksDetector> landm_net({}, {}, {});
+        cv::gapi::ie::Params<nets::FaceReidentificator> reident_net({}, {}, {});
+        cv::gapi::ie::Params<nets::PersonDetActionRec> action_net({}, {}, {});
+
+        ActionDetectorConfig action_config;
+        ActionDetectionKernelInput ad_kernel_input;
         if (!ad_model_path.empty()) {
             // Load action detector
-            ActionDetectorConfig action_config(ad_model_path);
-            action_config.deviceName = FLAGS_d_act;
-            action_config.ie = ie;
-            action_config.is_async = true;
+            action_config.net_with_six_actions = util::IsNetForSixActions(ad_model_path);
             action_config.detection_confidence_threshold = static_cast<float>(FLAGS_t_ad);
             action_config.action_confidence_threshold = static_cast<float>(FLAGS_t_ar);
             action_config.num_action_classes = actions_map.size();
-            action_detector.reset(new ActionDetection(action_config));
-        } else {
-            action_detector.reset(new NullDetection<DetectedAction>);
+            ad_kernel_input.ptr.reset(new ActionDetection(action_config));
+            std::array<std::string, 7> outputBlobList;
+            !action_config.net_with_six_actions ? outputBlobList = {"mbox_loc1/out/conv/flat",
+                                                                    "mbox_main_conf/out/conv/flat/softmax/flat",
+                                                                    "mbox/priorbox",
+                                                                    "out/anchor1",
+                                                                    "out/anchor2",
+                                                                    "out/anchor3",
+                                                                    "out/anchor4"} 
+                                                : outputBlobList = {"ActionNet/out_detection_loc",
+                                                                    "ActionNet/out_detection_conf",
+                                                                    "ActionNet/action_heads/out_head_1_anchor_1",
+                                                                    "ActionNet/action_heads/out_head_2_anchor_1",
+                                                                    "ActionNet/action_heads/out_head_2_anchor_2",
+                                                                    "ActionNet/action_heads/out_head_2_anchor_3",
+                                                                    "ActionNet/action_heads/out_head_2_anchor_4"};
```

Comment:
What does `net_with_six_actions` mean? Do models change from version to version?